src/Microsoft.ML.LightGbm/LightGbmArguments.cs

```diff
@@ -1,678 +1,225 @@
+        internal void UpdateParameters(Dictionary<string, object> res)
         {
-            internal const string Name = "gbdt";
-            internal const string FriendlyName = "Tree Booster";
-
-            /// <summary>
-            /// The options for <see cref="TreeBooster"/>, used for setting <see cref="Booster"/>.
-            /// </summary>
-            [TlcModule.Component(Name = Name, FriendlyName = FriendlyName, Desc = "Traditional Gradient Boosting Decision Tree.")]
-            public class Options : ISupportBoosterParameterFactory
+            FieldInfo[] fields = BoosterOptions.GetType().GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance);
+            foreach (var field in fields)
             {
-                /// <summary>
-                /// Whether training data is unbalanced. Used by <see cref="LightGbmBinaryClassificationTrainer"/>.
-                /// </summary>
-                [Argument(ArgumentType.AtMostOnce, HelpText = "Use for binary classification when training data is not balanced.", ShortName = "us")]
-                public bool UnbalancedSets = false;
-
-                /// <summary>
-                /// The minimum loss reduction required to make a further partition on a leaf node of the tree.
-                /// </summary>
-                /// <value>
-                /// Larger values make the algorithm more conservative.
-                /// </value>
-                [Argument(ArgumentType.AtMostOnce,
-                    HelpText = "Minimum loss reduction required to make a further partition on a leaf node of the tree. the larger, " +
-                        "the more conservative the algorithm will be.")]
-                [TlcModule.Range(Min = 0.0)]
-                public double MinimumSplitGain = 0;
-
-                /// <summary>
-                /// The maximum depth of a tree.
-                /// </summary>
-                /// <value>
-                /// 0 means no limit.
-                /// </value>
-                [Argument(ArgumentType.AtMostOnce,
-                    HelpText = "Maximum depth of a tree. 0 means no limit. However, tree still grows by best-first.")]
-                [TlcModule.Range(Min = 0, Max = int.MaxValue)]
-                public int MaximumTreeDepth = 0;
-
-                /// <summary>
-                /// The minimum sum of instance weight needed to form a new node.
-                /// </summary>
-                /// <value>
-                /// If the tree partition step results in a leaf node with the sum of instance weight less than <see cref="MinimumChildWeight"/>,
-                /// the building process will give up further partitioning. In linear regression mode, this simply corresponds to minimum number
-                /// of instances needed to be in each node. The larger, the more conservative the algorithm will be.
-                /// </value>
```

Comment:
what's happening to all these xml documentations? do you need to merge with master?

Reply:
thanks Shahab - I captured these.

---
In reply to: [265696202](https://github.com/dotnet/machinelearning/pull/2948#discussion_r265696202) [](ancestors = 265696202)